ams/server/src/main/java/com/netease/arctic/server/metrics/MetricsManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.metrics;
+
+import com.netease.arctic.ams.api.metrics.MetricsContent;
+import com.netease.arctic.ams.api.metrics.MetricsReporter;
+import org.apache.iceberg.metrics.MetricsReport;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Manage metrics and metrics reporter
+ * Register reporter by adding the following configuration in the system configuration:
+ * reporters:
+ *   - name: report1
+ *     impl: {reporter class}
+ *     properties:
+ *       host: 127.0.0.1
+ *       port: 8080
+ */
+public class MetricsManager {
+  private final Map<String, MetricsReporter> amoroReporters = new ConcurrentHashMap<>();
+  private final Map<String, org.apache.iceberg.metrics.MetricsReporter> icebergReporters = new ConcurrentHashMap<>();
+
+  public MetricsManager() {
+  }
+
+  public MetricsManager(List<ReporterMeta> metas) {
+    metas.forEach(this::register);
+  }
+
+  public void register(ReporterMeta meta) {
+    try {
```

Comment:
How about using SPI ?

Reply:
SPI can also be used to achieve this, but configuration is required when using the reporter in AMS. Even if SPI is used, it cannot reduce the configuration, but instead requires adding a mapping between the reporter type and reporter interface. In my opinion, using SPI in this scenario is unnecessary.WDYT.